### DISCUSSION TITLE
SQLModel maps Decimal to DECIMAL(10,0) in MySQL instead of respecting decimal_places

### DISCUSSION THREAD

**Discussion by jalvarezz13** [fastapi/sqlmodel#1127] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from decimal import Decimal

class TransactionBase(SQLModel):
    amount: Decimal = Field(default=0.0, gt=0.0, decimal_places=2)
```

### Description

SQLModel is mapping this field to DECIMAL(10,0) in MySQL, instead of DECIMAL(10,2) as expected. This results in loss of decimal precision in the database.

### Operating System

Linux, Other

### Operating System Details

MySQL Docker container using version 8.4.2

### SQLModel Version

0.0.22

### Python Version

3.12.3

### Additional Context

Currently, it's necessary to manually replace the field definition in the generated migration, changing DECIMAL(10,0) to DECIMAL(10,2) to achieve the desired behavior.

**Comment by loudblow** [comment]
Same issue!

**Comment by YuriiMotov** [comment]
Seems to work fine in current version (0.0.24):

```
op.create_table(
        'transactionbase',
        sa.Column('id', sa.Integer(), nullable=False),
        sa.Column('amount', sa.Numeric(scale=2), nullable=False),
        sa.PrimaryKeyConstraint('id')
    )
```
